src/data/matrix/kronecker.lean

```diff
@@ -221,7 +234,9 @@ lemma mul_kronecker_mul [comm_semiring α]
   (A ⬝ B) ⊗ₖ (A' ⬝ B') = (A ⊗ₖ A') ⬝ (B ⊗ₖ B') :=
 kronecker_map_linear_mul_mul (algebra.lmul ℕ α).to_linear_map mul_mul_mul_comm A B A' B'
 
--- insert lemmas specific to `kronecker` below this line
+lemma kronecker_assoc [comm_semiring α] (A : matrix l m α) (B : matrix n p α) (C : matrix q r α) :
+  reindex (equiv.prod_assoc l n q) (equiv.prod_assoc m p r) ((A ⊗ₖ B) ⊗ₖ C) = A ⊗ₖ (B ⊗ₖ C):=
```

Comment:
Is this easy to show for the tensor_product case too, as something like
```lean
  (reindex (equiv.prod_assoc l n q) (equiv.prod_assoc m p r) ((A ⊗ₖₜ[R] B) ⊗ₖₜ[R] C)).map tensor_product.assoc = A ⊗ₖₜ (B ⊗ₖₜ C) :=
```

Reply:
I though about doing so, it won't be hard. I can add it.

Reply:
Yep, it's easy:
```
lemma kronecker_tmul_assoc [semigroup α] (A : matrix l m α) (B : matrix n p α) (C : matrix q r α) :
  reindex (equiv.prod_assoc l n q) (equiv.prod_assoc m p r)
    (((A ⊗ₖₜ[R] B) ⊗ₖₜ[R] C).map (tensor_product.assoc _ _ _ _)) = A ⊗ₖₜ[R] (B ⊗ₖₜ[R] C):=
ext $ λ i j, assoc_tmul _ _ _
```

Reply:
I was rather trying
```
lemma kronecker_tmul_assoc (A : matrix l m α) (B : matrix n p β) (C : matrix q r γ) :
reindex (equiv.prod_assoc l n q) (equiv.prod_assoc m p r)
  ((A ⊗ₖₜ[R] B) ⊗ₖₜ[R] C).map tensor_product.assoc = A ⊗ₖₜ[R] (B ⊗ₖₜ[R] C) :=
```
with different types for A, B, C (but I still have to make a couple of compatibility work).

Reply:
Your version doesn't work because you have `reindex e1 e2 X.map Y` not `reindex e1 e2 (X.map Y)`

Reply:
Well, this does not work either
```
lemma kronecker_tmul_assoc (A : matrix l m α) (B : matrix n p β) (C : matrix q r γ) :
reindex (equiv.prod_assoc l n q) (equiv.prod_assoc m p r)
  ((A ⊗ₖₜ[R] B ⊗ₖₜ[R] C).map tensor_product.assoc) = A ⊗ₖₜ[R] (B ⊗ₖₜ[R] C) :=
```

Reply:
Right, because also `tensor_product.assoc` needs extra arguments as I added in https://github.com/leanprover-community/mathlib/pull/8700#discussion_r689552851

Reply:
Great, thanks! It worked. As far as `φ` is concerned, I am not sure to understand what you mean but as said I somewhat prefer the version with it as it seems more natural to me.

Reply:
whitespace nit:
```suggestion
  reindex (equiv.prod_assoc l n q) (equiv.prod_assoc m p r) ((A ⊗ₖ B) ⊗ₖ C) = A ⊗ₖ (B ⊗ₖ C) :=
```